frontend/src/pages/RunList.tsx

```diff
@@ -376,16 +376,42 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
    * DisplayRun will show '-'.
    */
   private async _getAndSetExperimentNames(displayRun: DisplayRun): Promise<void> {
-    const experimentId = RunUtils.getFirstExperimentReferenceId(displayRun.run);
-    if (experimentId) {
-      try {
-        const experiment = await Apis.experimentServiceApi.getExperiment(experimentId);
-        displayRun.experiment = { displayName: experiment.name || '', id: experimentId };
-      } catch (err) {
-        // This could be an API exception, or a JSON parse exception.
-        displayRun.error = 'Failed to get associated experiment: ' + await errorToMessage(err);
+    const experimentInfo = this._getFirstExperimentInfoInReferences(displayRun.run.resource_references);
+    if (experimentInfo) {
+      displayRun.experiment = experimentInfo;
+    } else {
+      const experimentId = RunUtils.getFirstExperimentReferenceId(displayRun.run);
```

Comment:
@IronPan Needs some clarification: can name be absent in experiment reference with new backend?

Open discussion: I thought it will always be there, so I preferred being simpler -- just get name from the reference.
Just getting name from reference is a backward-incompatible change considering just the frontend -- frontend no longer works with low version backend. I'm curious if that's something we need to be taking care of.

Thoughts?

Reply:
From my understanding when discussing with @IronPan, a name for a reference can be absent. Specifically, when a reference was created before thew changes made by @IronPan 
 in his recent [PR](https://github.com/kubeflow/pipelines/pull/1781). Checking for the name property within a reference should indicate whether a reference/backend was/is a lower version and mitigated and incompatibilities. When it is determined that an older reference or lower version backend is in use, the frontend will fallback to the previous method of fetching reference names. This allows for this change to be backwards compatible.

Reply:
Yes this is mostly for the consideration of existing data. 
alternatively we could backfill this to user database but that would be more complicated.

Reply:
Cool. I didn't know we need data backfill to make sure the field exists.
Then current solution sounds good to me and changing this single place is enough.